### PR TITLE
PHOENIX-6504 Restrict thirdparty guava imports to prevent RegionServer crash

### PR DESCRIPTION
**Pull request by virajjasani**

### CODE REVIEW

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 33s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ 4.16 Compile Tests _ |
| +0 :ok: |  mvndep  |   5m 36s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  13m 42s |  4.16 passed  |
| +1 :green_heart: |  compile  |   2m 35s |  4.16 passed  |
| +1 :green_heart: |  checkstyle  |   0m 59s |  4.16 passed  |
| +1 :green_heart: |  javadoc  |   2m 30s |  4.16 passed  |
| +0 :ok: |  spotbugs  |   6m 10s |  root in 4.16 has 999 extant spotbugs warnings.  |
| +0 :ok: |  spotbugs  |   3m 20s |  phoenix-core in 4.16 has 944 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 20s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |  13m 26s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 30s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 30s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 57s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  2s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   2m 25s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |  10m 35s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 153m 47s |  root in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 20s |  The patch does not generate ASF License warnings.  |
|  |   | 221m 18s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | phoenix.pherf.PherfMainIT |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1260/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1260 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile xml |
| uname | Linux c37892eb8204 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | 4.16 / 49ff48a |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1260/1/artifact/yetus-general-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1260/1/testReport/ |
| Max. process+thread count | 5661 (vs. ulimit of 30000) |
| modules | C: phoenix-core . U: . |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1260/1/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 30s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ 4.16 Compile Tests _ |
| +0 :ok: |  mvndep  |   5m 17s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  13m 23s |  4.16 passed  |
| +1 :green_heart: |  compile  |   1m 47s |  4.16 passed  |
| +1 :green_heart: |  checkstyle  |   0m 49s |  4.16 passed  |
| +1 :green_heart: |  javadoc  |   1m 57s |  4.16 passed  |
| +0 :ok: |  spotbugs  |   4m 23s |  root in 4.16 has 999 extant spotbugs warnings.  |
| +0 :ok: |  spotbugs  |   2m 55s |  phoenix-core in 4.16 has 944 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   9m 50s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 45s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 45s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 50s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  1s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   1m 54s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   7m 47s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 149m  9s |  root in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 21s |  The patch does not generate ASF License warnings.  |
|  |   | 204m 25s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | phoenix.pherf.PherfMainIT |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1260/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1260 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile xml |
| uname | Linux ae455238e7eb 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | 4.16 / 49ff48a |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1260/2/artifact/yetus-general-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1260/2/testReport/ |
| Max. process+thread count | 5975 (vs. ulimit of 30000) |
| modules | C: phoenix-core . U: . |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1260/2/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: pom.xml [hunk 1]
```diff
@@ -420,6 +420,13 @@
                       <bannedImport>org.apache.commons.logging.LogFactory</bannedImport>
                     </bannedImports>
                   </restrictImports>
+                  <restrictImports implementation="de.skuzzle.enforcer.restrictimports.rule.RestrictImports">
+                    <includeTestCode>true</includeTestCode>
+                    <reason>Do not use phoenix-thirdparty provided Guava</reason>
```
Comment (stoty) [inline pom.xml:425]: Nit: maybe emphasise that's it's for 4.x only.

**Review by stoty — APPROVED** [review]
+1 LGTM

File: pom.xml (hunk 1 above)
Comment (virajjasani) [reply pom.xml:425]: Yeah, makes sense. Thanks

**Pull request merged**